var/spack/repos/builtin/packages/intel-parallel-studio/package.py

```diff
@@ -145,373 +145,71 @@ class IntelParallelStudio(IntelPackage):
     variant('vtune',     default=False,
             description='Install the Intel VTune Amplifier XE')
 
-    provides('daal', when='+daal')
-
-    provides('ipp', when='+ipp')
+    provides('daal',        when='+daal')
+    provides('ipp',         when='+ipp')
 
-    provides('mkl',       when='+mkl')
-    provides('blas',      when='+mkl')
-    provides('lapack',    when='+mkl')
-    provides('scalapack', when='+mkl')
+    provides('mkl',         when='+mkl')
+    provides('blas',        when='+mkl')
+    provides('lapack',      when='+mkl')
+    provides('scalapack',   when='+mkl')
 
-    provides('mpi', when='+mpi')
+    provides('mpi',         when='+mpi')
+    provides('tbb',         when='+tbb')
 
-    provides('tbb', when='+tbb')
+    # For TBB, static linkage is not and has never been supported by Intel:
+    # https://www.threadingbuildingblocks.org/faq/there-version-tbb-provides-statically-linked-libraries
+    conflicts('+tbb',       when='~shared')
 
-    # The following components are not available in the Composer Edition
     conflicts('+advisor',   when='@composer.0:composer.9999')
     conflicts('+clck',      when='@composer.0:composer.9999')
     conflicts('+inspector', when='@composer.0:composer.9999')
     conflicts('+itac',      when='@composer.0:composer.9999')
+    conflicts('+mpi',       when='@composer.0:composer.9999')
```

Comment:
one problem with the +mpi conflicts is that the default variant is set to +mpi, so users will get an error if they try to install the professional or composer editions, unless they explicitly request ~mpi. Ideally, installing the cluster edition would default to +mpi, while the professional and composer editions would default to ~mpi, but I don't know of any mechanisms in Spack to make that happen. In the absence of this ideal solution, I feel like ~mpi should be the default, that way by default all versions will install.

Reply:
@lee218llnl 
I have been caught by this as well, and [wondered about conditional defaults](https://github.com/spack/spack/pull/7469#issuecomment-380981224).

I fear `~mpi` as default will set silent and unpleasant traps if a user installs the cluster edition with defaults: As soon as the virtual `mpi` is called for, either Spack triggers a "duplicate" install of `intel-parallel-studio`, or the user might wonder why Spack still insists on OpenMPI.

By constrast, as @mkuron [mentioned](https://github.com/spack/spack/pull/7469#issuecomment-381077622
) (my emphasis):
> Personally, I'd be fine with leaving the problem as-is for now. If you try to spack install intel-parallel-studio@composer.2015.1, **it tells you that the default variant has a conflict**, so you immediately know you need to do spack install intel-parallel-studio@composer.2015.1 -daal -intel instead …

Because of that better transparency, I am inclined to leave it as-is, too. What do you say?

Reply:
OK, I can see the merits in both sides and I don't have a strong opinion on this, so I'm OK with it as is.